exporter/elasticsearchexporter/bulkindexer_test.go

```diff
@@ -194,3 +194,68 @@ func TestNewBulkIndexer(t *testing.T) {
 	bi := newBulkIndexer(client, cfg.(*Config), true, nil, nil)
 	t.Cleanup(func() { bi.Close(t.Context()) })
 }
+
+func TestGetErrorHint(t *testing.T) {
+	tests := []struct {
+		name      string
+		index     string
+		errorType string
+		want      string
+	}{
+		{
+			name:      "version_conflict_engine_exception with .ds-metrics- prefix",
+			index:     ".ds-metrics-foo",
+			errorType: "version_conflict_engine_exception",
+			want:      "check the \"Known issues\" section of Elasticsearch Exporter docs",
+		},
+		{
+			name:      "illegal_argument_exception with .otel in index (OTel mapping mode)",
+			index:     "logs-generic.otel-default",
+			errorType: "illegal_argument_exception",
+			want:      "This error typically occurs when using OTel mapping mode (default from v0.122.0) with Elasticsearch < 8.12. OTel mapping mode requires Elasticsearch 8.12+ (ideally 8.16+). To resolve: upgrade Elasticsearch to 8.12+, or use a different mapping mode (e.g., mapping::mode: ecs or mapping::mode: raw). See the README for more details.",
```

Comment:
nit: The two expected strings used for `want` could be a local variable so it's not repeated over and over again in test cases.